kucher/view/main_window/register_view_widget/import_export_dialog.py

```diff
@@ -0,0 +1,310 @@
+    file_name, _ = QFileDialog.getSaveFileName(dialog_box, 'Export configuration file', 'config.yml',
+                                               'YAML Files (*.yml  *.yaml);;All Files (*)')
+    if not file_name:
+        return
+    try:
+        _file = open(file_name, 'w+')
+
+    except Exception as ex:
+        show_error('Export failed',
+                   f'Cannot export {file_name}',
+                   f'Error: {ex}',
+                   parent)
+        _logger.exception(f'File {file_name} could not be exported: {ex!r}')
+        return
+
+    async def executor():
+        register_yaml = {}
+        for reg in registers:
+            if reg.mutable and reg.persistent:
+                try:
+                    register_yaml[reg.name] = await reg.read_through()
+                except Exception as ex:
+                    show_error('Export failed',
+                               f'Cannot export {file_name}',
+                               f'This parameter cannot be read:\n{reg.name}',
+                               parent)
+                    _logger.exception(f'Register {reg} could not be read')
+                    return
+
+        yaml.dump(register_yaml, _file)
+        _file.close()
+
+    asyncio.get_event_loop().create_task(executor())
+
+    display_sucess_message('Export successful',
+                           f'Parameters have been successfully exported to:\n{file_name}',
+                           '',
+                           parent)
+
+
+def import_registers(parent, registers):
+    """
+    Import registers values from a .yml (or .yaml) file.
+    The file must match the following pattern:
+
+    full_register_name:
+    - register_value
+
+    e.g:
+
+    ctl.num_attempts:
+    - 100
+    ctl.pwron_slftst:
+    - false
+    ctl.spinup_durat:
+    - 1.7000000476837158
+
+    It can only contain available registers on the current device, with values within the allowed range. The values
+    must be the right type and dimension. If a register is not mentioned in the file, it will keep its former value.
+    """
+    dialog_box = QWidget()
+    dialog_box.setGeometry(10, 10, 1000, 700)
+    file_name, _ = QFileDialog.getOpenFileName(dialog_box, 'Import configuration file', '',
+                                               'YAML Files (*.yml *.yaml);;All Files (*)')
+    if not file_name:
+        return
+    try:
+        file = open(file_name, 'r')
+    except IOError:
+        _logger.exception(f'File {file_name} could not be open')
+        return
+
+    imported_registers = yaml.load(file, Loader=yaml.Loader)
+
+    if imported_registers:
+        check_passed = check_registers(parent, file_name, file, registers, imported_registers)
+        if check_passed:
+            write_registers(parent, file_name, registers, imported_registers)
+    else:
+        write_registers(parent, file_name, registers, imported_registers)
+
+
+def check_registers(parent, file_name, file, registers, imported_registers, ):
```

Comment:
The first three parameters are not needed. This function violates the principle of separation of concerns; its job is to check the correctness, not to display error messages or close files. All it needs is the register model representation (`registers` here) and the set of imported registers. Report the result by returning an error code (define an enumeration), then handle it in the caller. Also, the trailing comma is not needed.